gufe/protocols/protocol.py

```diff
@@ -111,6 +116,17 @@ def __init__(self, settings: Settings):
         Once the Protocol object is created, the input Settings are frozen,
         so should be finalised before creating the Protocol instance.
         """
+
+        if not hasattr(self.__class__, "_settings_cls"):
```

Comment:
A thought for another issue, we don't do this for `self._settings` or `results_cls`. Would it be worth eventually cleaning up all these abstract classes with expected attributes by adding `__init_subclass__` methods?

Reply:
@IAlibay can you elaborate on this in an issue?